app.js

```diff
@@ -14,16 +13,21 @@ connect('http://127.0.0.1:8088', 'user', 'pass')
   .done();
 
 /**
- * Starts Stasis app 'sla' and initiates SLA application after ARI connection.
+ * Waits for a StasisStart event before going into the main SLA module
  * @param {Object} client - Object that contains information from the ARI 
  *   connection.
  */
 function clientLoaded (client) {
   client.start('sla');
-  sla(client)
-    .then(console.log)
-    .catch(errHandler)
-    .done();
+  client.on('StasisStart', function(event, channel) {
+    if(event.args[0] !== 'dialed') {
+      var bridgeName = event.args[0];
+      sla(client, channel, bridgeName)
+        .then(console.log)
+        .catch(errHandler)
```

Comment:
You should handle different errors differently. Not all errors thrown from sla should necessarily result in your program blowing up.